azure-mgmt-containerregistry/src/main/java/com/microsoft/azure/management/containerregistry/Registry.java

```diff
@@ -37,55 +40,146 @@
      */
     DateTime creationDate();
 
     /**
-     * @return the value that indicates whether the admin user is enabled. This value is false by default.
+     * @return the value that indicates whether the admin user is enabled
      */
     boolean adminUserEnabled();
 
     /**
-     * @return the name of the storage account for the container registry.
+     * @return the name of the storage account for the container registry; 'null' if container register SKU a managed tier
      */
     String storageAccountName();
 
     /**
-     * @return the login credentials for the specified container registry.
+     * @return the ID of the storage account for the container registry; 'null' if container register SKU a managed tier
+     */
+    @Beta(SinceVersion.V1_4_0)
+    String storageAccountId();
+
+    /**
+     * @return the login credentials for the specified container registry
      */
-    RegistryListCredentials listCredentials();
+    @Beta(SinceVersion.V1_4_0)
+    RegistryCredentials getCredentials();
 
     /**
-     * @return the login credentials for the specified container registry.
+     * @return a representation of the future computation of this call
      */
-    Observable<RegistryListCredentials> listCredentialsAsync();
+    @Beta(SinceVersion.V1_4_0)
+    Observable<RegistryCredentials> getCredentialsAsync();
 
     /**
      * Regenerates one of the login credentials for the specified container registry.
-     * @param passwordName the password name
+     * @param accessKeyType the admin user access key name to regenerate the value for
      * @return the result of the regeneration
      */
-    RegistryListCredentials regenerateCredential(PasswordName passwordName);
+    @Beta(SinceVersion.V1_4_0)
+    RegistryCredentials regenerateCredential(AccessKeyType accessKeyType);
 
     /**
      * Regenerates one of the login credentials for the specified container registry.
-     * @param passwordName the password name
-     * @return the result of the regeneration
+     *
+     * @param accessKeyType the admin user access key name to regenerate the value for
+     * @return a representation of the future computation of this call
+     */
+    @Beta(SinceVersion.V1_4_0)
+    Observable<RegistryCredentials> regenerateCredentialAsync(AccessKeyType accessKeyType);
+
+    /**
+     * Lists the quota usages for the specified container registry.
+     *
+     * @return the list of container registry's quota usages
+     */
+    @Beta(SinceVersion.V1_4_0)
+    Collection<RegistryUsage> listQuotaUsages();
+
+    /**
+     * Lists the quota usages for the specified container registry.
+     *
+     * @return a representation of the future computation of this call
+     */
+    @Beta(SinceVersion.V1_4_0)
+    Observable<RegistryUsage> listQuotaUsagesAsync();
+
+    /**
+     * @return returns entry point to manage container registry webhooks.
+     */
+    WebhookOps webhooks();

```

Comment:
so in places like this, where the parent provides access to a special collection of its children we just call that type the plural of the child type, so here that would simply be `Webhooks`, not `WebhookOps`. Any particular reason or this discrepancy?

Reply:
I renamed it to WebhookOperations/WebhookOperationsImpl since we already have a WebhooksImpl and overloading that name can lead to confusion when looking for implementations of it.